openapi.yaml

```diff
@@ -2165,6 +2177,634 @@ paths:
+          description: Catalog ID to delete
+          required: true
+          schema:
+            type: string
+      responses:
+        400:
+          description: Invalid ID supplied
+          content: {}
+        404:
+          description: Catalog not found
+          content: {}
+      security:
+        - oscal_auth:
+            - write:catalogs
+            - read:catalogs
+  /plan-of-action-and-milestones/search:
+    get:
+      tags:
+        - OSCAL Plan of Action and Milestones
+      summary: Searches for OSCAL Plan of Action and Milestones
+      operationId: findPoamsByName
+      parameters:
+        - name: query
+          in: query
+          description: Terms to search for in catalog names
+          required: true
+          schema:
+            type: string
+      responses:
+        200:
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/OSCALCatalog"
+        400:
+          description: Invalid status value
+          content: {}
+      security:
+        - oscal_auth:
+            - read:catalogs
+
+  /system-assessment-plans:
+    get:
+      tags:
+        - OSCAL System Assessment Plans
+      summary: Returns all OSCAL System Assessment Plans
+      operationId: getAssessmentPlan
+      responses:
+        200:
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/OSCALCatalog"
+      security:
+        - oscal_auth:
+            - read:catalogs
+    post:
+      tags:
+        - OSCAL System Assessment Plans
+      summary: Adds a new OSCAL System Assessment Plan
+      operationId: addAssessmentPlan
+      requestBody:
+        description: OSCAL System Assessment Plans to be added
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/OSCALCatalog"
+          multipart/form-data:
+            schema:
+              type: object
+              properties:
+                url:
+                  type: string
+                  format: binary
+                fileName:
+                  type: string
+                  format: binary
+        required: true
+      responses:
+        405:
+          description: Invalid input
+          content: {}
+      security:
+        - oscal_auth:
+            - write:catalogs
+            - read:catalogs
+      x-codegen-request-body-name: body
+    /system-assesment-plans/{catalogId}:
+      get:
+      tags:
+        - OSCAL System Assessment Plans
+      summary: Finds an OSCAL System Assessment Plans catalog by ID
+      description: Returns a single OSCAL System Assessment Plan
+      operationId: getAssessmentPlanById
+      parameters:
+        - name: catalogId
```

Comment:
Swagger Editor is also giving errors on corresponding `{catelogId}` in the `/system-assessment-plans` path. Is this not already part of the path as shown in line 2439 `/system-assessment-plans/{catalogId}`?

Reply:
Can you post the actual error?

Reply:
Would an `getAssessmentPlanById` even use a `catalogId`?